cmake/options.cmake

```diff
@@ -146,7 +146,7 @@ endif()
 
 add_library(svs_native_options INTERFACE)
 add_library(svs::native_options ALIAS svs_native_options)
-target_compile_options(svs_native_options INTERFACE -march=native -mtune=native)
+target_compile_options(svs_native_options INTERFACE -DSVS_CPUARCH_NATIVE -march=native -mtune=native)
```

Comment:
If I understand correctly, `SVS_CPUARCH_NATIVE` should be an option similar to the other options, such as `SVS_NO_AVX512`. By default, it can be set to `OFF`. In this case, it will compile for all `x86_64` architectures if building on an x86 platform.